### DISCUSSION TITLE
Drag and Drop elements in touch screens (Editing on mobile device) (Migrated from #482)

### DISCUSSION THREAD

**Discussion by grrog** [elementor/elementor#23528] [Feature Request]
Hello,

I tried to edit some pages on the go on my mobile (Opera mobile browser, last version) and I beleive Elementor don't recognise the "touch" action.

Is there a plan to make Elementor editing compatible with mobile devices ?

Elementor is a very great live page builder! Congrats to all the team behind this plugin!

**Comment by vsimon51** [comment]
+1. I agree.Makes a bit of a nonsense of the WYSIWYG nature of Elementor to have to switch to a view you are not editing in order to move, duplicate, or save widgets.

**Comment by jaffa2** [comment]
Hi, thanks for linking, the status of this one looked unassigned, so I created a new one apologies.

Is there any chance of this happening?

Many thanks

**Comment by LogoLogics2** [comment]
I agree, I work on a touch screen 27 inch All in one and use the touch screen for nearly every action, even while editing in Affinity Photo and Designer, and can fully work with my touch screen.

I was surprised to find that the Elementor edit mode does not recognize the touch moves, because apart from that, Elementor has proven to be outstanding and up to date, maybe even, ahead of lots of new techniques.

I hope this can be achieved at some point_

**Comment by danielrehmann** [comment]
I also think that it should be able to drag in sections and widgets via touch screen. It's very convenient to change something with a mobile device quickly without having to have a pc around.

**Comment by gracevillarino** [comment]
Me and friend would like to be able drag and drop elements on a touch screen. Checked Divi theme and is having the same problem on touchscreens.
2 votes

**Comment by nvmom** [comment]
I'd like to echo everyone's comments here. The web is moving to mobile, and it's very unfortunate that we can't use Elementor on our mobile devices. Would love to see this changed!

**Comment by cbrcongit** [comment]
Surely there's JS scripts useful to get stylus at least?

(Any links/suggestions?...as a simple addon plugin for touch device users?)

... I think I'm only missing "double-click" and "right click"  (to use any drag&drop for blocks, (or click&drop for mobile?) )

 ![No mouse warning](http://imgur.com/gallery/jSlYxEY)

**Comment by Kartik4211** [comment]
We all my friends feel disappointed as you remove  your drag and drop features on mobile phones and now we are not able to operate elementor on mobile . So we all requests to company that it again bring drag and drop features on mobile too.

**Comment by Kartik4211** [comment]
We all will be thankful to company if we will able to use elementor on mobiles with touch screen

**Comment by Kartik4211** [comment]
We are waiting for when elementor recognise drag and drop options on mobile. So please company respond to it as soon as possible.

**Comment by mercury168** [comment]
Yes i agree. This is badly needed. I have requested for this feature so many times already, and it seems elementor is not interested to develop it. I think at the minimum, horizontal and vertical scroll bars for the screens, and then a reaponsive single column vertical scroll left element tab.....for the meantime, at least, while they are developing the code for it. Its such a hassle to not be able do basic stuff like to correct spelling or grammar on the go thru mobile at the very least or to fix the site while you are on a crowded bus or train where a laptop or tablet wont work.

**Comment by danielrehmann** [comment]
This issue is really important for me. I am actually thinking about ditching elementor Pro in the near future if the support for mobile editing will not improve.

**Comment by lucasbols** [comment]
+1 on this!

**Comment by incompletebiped** [comment]
+1 for the win! I use Elementor on multiple sites but being able to make on the fly edits via mobile would really put Elementor over the top. I'd pay for the ability. An app would be nice.

**Comment by envyzweck** [comment]
+1 on this!

**Comment by maxie71** [comment]
+1

**Comment by brassben** [comment]
+1

**Comment by hotelpress** [comment]
What about using chrome on a surafce pro? WIndows 10 home. 

With or without the keyboard/trackapd attached?

**Comment by SimonEvans482** [comment]
+1 I'll add my name to this too. My laptop went 'bang' and now i'm doing everything on a touch-screen; touch-scren support from elementor would be splendid. Any word from the developers on this?

**Comment by grrog** [comment]
Hey, ok I see many people want an "on the go" editing same to me 3 years ago so… it's time to do something Elementor's team, no ? (I'm with elementor pro for 2 years now : I love this  Wordpress Elevator! Please make this :)

**Comment by quentintricoire** [comment]
Please for the next version ? Make Elementor more powerful !

**Comment by ghost** [comment]
This absolutely needs to be a feature! My main computer is an iPad Pro and it is very frustrating not to be able to edit my Elementor website on it.

**Comment by TomDowling** [comment]
+1 for this feature, need to be able to work on the go, and I only take my iPad now.

**Comment by MaximPro** [comment]
This is probably one of the most Epic Fails I've come accross lately.
Elementor is trying to make Webediting easy... wich is good.
HOWEVER IT IS NOT POSSIBLE TO WORK ON AN WEBPAGE WITH AN TABLET ??? WTF !!!
2019 and still doesn't work, tryed with 2 Browsers on 2 different Tablets.

Try it for yourself or urgently implement a different way to add elements, except for drag and drop.
doubleclick for example to add to the current container...
and maybe add an extra tablet mode, if drag and drop isn't easy.

This should be a high priority process!!
WTF Tousands of People cannot edit their Elementor Webpage with an Tablet or an Smartphone.
Probably this will be on some Headlines soon... if this is not yet fixed.

Sry for the harsh wording, but how IS THIS POSSIBLE after 3 Years of developement?

FIX IT ASAP!

**Comment by hotelpress** [comment]
One response I got was that mobile browsers arnt powerful enough...

The new iPad pro benchmarks are 2x to 3x higher than my current MacBook :)

Plus, tablet's such as surface also have i5 and i7 processors now. 

Even my phone is ridiculously powerful.

C'mon... this will be a game changer and appeal to millions of users.

One update... I have a new Surface Go. While Elementor doesnt work in Edge, it does work on Chrome.

Touch screen no... but when I have the keyboard & trackpad attached it works fine.

I would love for it to work with a stylus like the surface pen... haven't tried yet but will soon.

This device is beautiful for taking with me anywhere I go and I can fix and design literally anywhere.

**Comment by 3sideslive** [comment]
Have to add my name to the list. I was stunned to discover I can't use Elementor on my tablet.  It's kind of a deal-breaker at this point, to be honest - I spend so much time working on my tablet that now I have to decide whether to continue investing time and money, or just drop it. I'm certainly not going to give up my tablet and park myself at my desk all night in order to make it work. Either it will become obvious soon that this is a priority and on its way, or I'll head in another direction soon. It's great, admittedly, but just not worth the aggravation.

**Comment by rushter777** [comment]
@arielk A simple response about this would be fantastic.

**Comment by hank3rd** [comment]
I am playing around with iPadOS 13 developer beta and I have the functionality to edit in Elementor on "Desktop Class" Safari! YAY!!!!

**Comment by rushter777** [comment]
I was on the Apple beta with my iPad but have not received anything for iPadOS 13 yet.

**Comment by hank3rd** [comment]
> I was on the Apple beta with my iPad but have not received anything for iPadOS 13 yet.

It is a developer beta, released Monday. Public beta is coming next month.  :)

**Comment by rushter777** [comment]
Ah yes. So send me your iPadOS profile.

**Comment by hank3rd** [comment]
> Ah yes. So send me your iPadOS profile.
> […](#)

Not sure what you mean. You need either be on a developer account or wait for the public beta.

**Comment by Dacobah** [comment]
Any news about a possible mobile feature? The Elementor editing support on the dashboard from a mobile device or a tablet would be very much appreciated.

Furthermore it seems I'm not the only one who is interested this feature :)

**Comment by hank3rd** [comment]
@Dacobah Not sure if you noticed my earlier posts, but Safari on the iPad in iOS 13 will allow for editing since it will be a "desktop class" browser. I have been using the beta for some time now and it works well.

**Comment by rushter777** [comment]
I remember you mentioning that before and I will test it out @hank3rd . You
can also use a mouse now with iPadOS if that helps @dacobah

**Comment by ghost** [comment]
That only helps if you are an iphone/iPad user..I'm not unfortunately

**Comment by hfiv3** [comment]
@hank3rd I’m using The iPad iOS 13 but I have to say that Elementor is not really working. I cannot scroll down because It will move sections. Do you have the same problem or is there a fix for this? Having this problem with more services.

**Comment by hank3rd** [comment]
I am able to scroll, you just need to be careful where you touch.

**Comment by hfiv3** [comment]
Allright.. I really hoped for a better experience.

Op wo 10 jul. 2019 om 00:53 schreef hank3rd <notifications@github.com>

**Comment by Dcfranklyn** [comment]
@hank3rd
To confirm: using IpadOS within safari browser, your having full editor functions? 
- drag & drop elements and sections
- add new elements
- edit elements
- right click elements to copy & paste them
- etc
???????

I just want to double check because I'm planning on buying an IPad Pro shortly and this would be a deal breaker if it didn't fully work.

**Comment by hank3rd** [comment]
> @hank3rd
> To confirm: using IpadOS within safari browser, your having full editor functions?
> 
> * drag & drop elements and sections
> * add new elements
> * edit elements
> * right click elements to copy & paste them
> * etc
>   ???????
> 
> I just want to double check because I'm planning on buying an IPad Pro shortly and this would be a deal breaker if it didn't fully work.

I have only tried to edit some pages, but I went back to test the items on your list. I was able to do everything except the right click copy/paste. I might be missing something though.

**Comment by Dcfranklyn** [comment]
@hank3rd
Thank you so much for taking the time to test this out for me. I appreciate it.

I only need the iPad for basic edits when I'm mobile. So having 95% functionality is good enough.

The right click might require a long press or a mouse attached.. who knows.

Thank you again

**Comment by hank3rd** [comment]
I tried a USB mouse and the “right click” did not work.

**Comment by hfiv3** [comment]
To be honest, it’s not really working well. Unable to ‘scroll’ down.
Accidentally moving objects all the time. Was able to edit a current
content ok.

Op za 20 jul. 2019 om 23:20 schreef hank3rd <notifications@github.com>

**Comment by Dcfranklyn** [comment]
@rogiervwagtendonk @hank3rd 

I wanted to come back and list my experience for others with the same problem.

I picked up an IPad Pro yesterday and installed the IpadOS on it. After doing some testing with Elementor on a clients website I found it’s very possible to do my work on mobile when needed.

- I found attaching a bluetooth mouse allowed me to completely avoid the scrolling / accidental element moving problem all together. (The mouse does have a learning curve & setup process but I got used to this quickly)
- Adding new elements required a short click and hold
- Moving sections & columns are done as normal
- Moving elements required a click and hold on the pencil edit icon
- Editing specs of a element, section, column are all done as normal
- Only thing I didn’t figure out yet is turning an section into a template

The only issues I ran into and how I fixed them:

Problem 1: You’re unable to right click on the iPad, so right clicking elements to use the menu options such as copy and paste styles wasn’t there. 

Fix 1: I looked up [elementor hotkeys](https://docs.elementor.com/article/105-hotkeys) and this solved the problem using my Bluetooth keyboard. (Only works in safari from my testing. Hold option + cmd within elementor to see the hotkeys)

-
Problem 2: Both the delete key & right click to delete wasn’t working on the iPad for elementor either. This was a problem when trying to delete a element that you have placed.

Fix 2: I couldn’t find a way to change the hotkey so I made a loop hole (not the best but it works).
I would create a new sections, drag the element I want to delete into the section, then delete the entire section. This isn’t the best solution but considering I found an answer to all the other interface problems.. I can deal with this when on the go.

In conclusion, during my testing period I found the IPad using IpadOS w/ a Bluetooth mouse & keyboard was able to handle 95% of the elementor edit interface concerns that I first had. I hope this is hopeful to someone. :D

**Comment by hfiv3** [comment]
Thanks so much for sharing your experience. Much better then mine, will
test it out again with an bluetooth mouse this time.

Best Rogier

Op zo 21 jul. 2019 om 20:47 schreef Dcfranklyn <notifications@github.com>

**Comment by mehran92** [comment]
Hi guys, i just managed to get scrolling working with the new OS 13 Beta 3 update. 

1. **Install OS 13 Beta :** go here to download, https://beta.apple.com/sp/betaprogram/ here is a video instruction on how to install the new update https://www.youtube.com/watch?v=0AomDpTfZnw

2. **Setup wireless mouse :** I am using a Logitech MX Master 2S (no dongle needed). Here is a video also on how to connect your mouse https://www.youtube.com/watch?v=OTEDQEhglVA

3. **Request desktop site:** on Safari you have to click the aA beside the address bar, on Chrome you can get it by hitting the setting icon on the right site of the browser.

Once that is done, you can scroll and use a mouse. Even though the scroll wheel works on other apps and website, it won't work for some reason on elementor edit mode yet. So, you will have to hover over the scrollbar and click and drag like you would on your computer (move the mouse all the way to the right side of the screen). I hope this helps, it wont be as smooth as working on a PC, but at least you can work on it now :)

**Comment by Nachdenker** [comment]
@hotelpress Have you tried editing your site with an iPad AND a Pencil?
Could the Surface Pencil improve anything?

Can someone tell me, if you are able to edit content (Write and edit text) on an iPad?
If I were - at least - able to edit existing content with and iPad (and a Pencil) it would be great.
And for now enough, till an iPad App has been launched by Elementor.

@shilo-ey For #Elementor it would be the next Milestone, to enable Website-Creation & Editing on mobile devices. I am sure you'll achieve this - together with us.

**Comment by bmx269** [comment]
Mobile / Tablet (touch) editing would be a great feature. Using a library like Modernizer to detect features, and target around that would definitely help start that process.

**Comment by Netvilox** [comment]
Ggghhhhhhu[Netvilox](http://netvilox.com) Ijjjjvgcfsedcbj

**Comment by Jenn0014** [comment]
Agree with all. Editing with mobile touch screen is a must! Pretty please

**Comment by imPrathamDev** [comment]
> Hello,
> 
> I tried to edit some pages on the go on my mobile (Opera mobile browser, last version) and I beleive Elementor don't recognise the "touch" action.
> 
> Is there a plan to make Elementor editing compatible with mobile devices ?
> 
> Elementor is a very great live page builder! Congrats to all the team behind this plugin!

I used elementor in many different web browsers now I can say that you can use elementor in android phone by using puffin browser. Simply use mouse option in puffin browser.

**Comment by hank3rd** [comment]
I just tried using the Apple Trackpad 2 with my iPad Pro and the scrolling with Elementor is much better and allows for better editing. We are getting there!

**Comment by hfiv3** [comment]
Ah yes great idea actually to use the trackpad. Also pretty portable. But
little strange to use it when your screen is touchscreen though.

Op wo 13 mei 2020 om 22:55 schreef hank3rd <notifications@github.com>

**Comment by Dcfranklyn** [comment]
Hello Everyone,

It's been almost a year since my last post and I thought I would give a update since the new mouse/trackpad updates have been made to the Ipad.

So I'm going to edit my previous points as they apply today

I found attaching a bluetooth mouse allowed me to completely avoid the scrolling / accidental element moving problem all together. (The mouse does have a learning curve & setup process but I got used to this quickly)

- The new curse, hover, and trackpad support has made the overall experience much better then before. Everything is very smooth for the most part. It's still not 100% perfect but it's closer. 

- Using the trackpad this time. I had no issues scrolling at all. The full "desktop" page wasn't being shown within the edit window. So I did have to zoom out a bit to correct that. (Side scrolling with a finger works as well)


-------
Adding new elements required a short click and hold


- With the trackpad, a normal click and drag works now. Didn't need to click and hold

-------
Moving sections & columns are done as normal

- Still working well
-------

Moving elements required a click and hold on the pencil edit icon

- With the trackpad and the new hover support. I was able to drag and drop as I do on a desktop computer

- You can now hover over the pencil and the copy option will reveal itself.

----

Editing specs of an element, section, column are all done as normal

- No changes

-----
Only thing I didn’t figure out yet is turning an section into a template

- The hover options are now available.  Along with right-click options (2 finger tap). So this is working now

-----

**You will need a mouse or Trackpad - Finger/touch still doesn't work**
I'm not sure about the controls within WordPress backend, but within Elementor, everything seems to be working great now. I use my Ipad as an on-to-go machine. I would say it's possible to make edits on a clients website now.

**Comment by naitsirhc007** [comment]
I have a Samsung Galaxy Note 8 (and really with any Android) I think we can edit using elementor.

It's just for urgently changes, on the go!!

1. Download Puffin Browser.
2. Edit a web site using mobile view
3. Puffin Browser comes with a scroll mouse integrated and no need a external mouse, just you use your fingers.

Note: you have to read and know how to use some gestures, to scroll, to pick, drag and drop, etc.

You have must use hold tap to right click, move elements with 2 fingers, you must try by yourself, and you can do step by step...

I have a tutorial video, but I have really busy to complete the edition and upload to YouTube, I hope very soon I can do that.

Cheers!

Hackerloko.

**Comment by desis123** [comment]
We badly need this feature...

**Comment by swhn** [comment]
> I have a Samsung Galaxy Note 8 (and really with any Android) I think we can edit using elementor.
> 
> It's just for urgently changes, on the go!!
> 
> 1. Download Puffin Browser.
> 2. Edit a web site using mobile view
> 3. Puffin Browser comes with a scroll mouse integrated and no need a external mouse, just you use your fingers.
> 
> Note: you have to read and know how to use some gestures, to scroll, to pick, drag and drop, etc.
> 
> You have must use hold tap to right click, move elements with 2 fingers, you must try by yourself, and you can do step by step...
> 
> I have a tutorial video, but I have really busy to complete the edition and upload to YouTube, I hope very soon I can do that.
> 
> Cheers!
> 
> Hackerloko.

I can't load Edit Page with puffin browser even turned on safe mode. It's OK to load on other browser but can't drag.

**Comment by atar-axis** [comment]
4 years later. Paid a ton of money for features I absolutely never use and such a basic feature is still missing. What a pitty :(

**Comment by androidacy-user** [comment]
The editor seems to "work" on mobile, it's just extremely difficult to navigate because no effort has been made to make the editor responsive - really a pity
 
Sure as heck not buying pro if free lacks even that basic thing

**Comment by ConsultMePL** [comment]
It's just amazing it's 2021 and still not working, and no one from Elementor doesn't give a shit.

For me, it's not a matter of touch (although it would be nice), but I have Samsung Galaxy Tab S7+ with keyboard and touchpad. But neither with a touchpad nor with a proper bluetooth mouse, I can use drag & drop. It's because Elementor doesn't enable drag & drop for mobile devices.

Yes, it somehow works in puffin browser, but only somehow - very laggy and impossible to use it on the regular basis.

Has anyone found a browser for Android that would support drag and drop in Elementor using either touchpad or bluetooth mouse?

**Comment by grrog** [comment]
Yes, I started this discussion FIVE years ago and I'm disgusted that nothing has changed since, despite Elementor's success. 

It's really unfortunate, that poses a question to me.

And yet I switched to Elementor pro… Hello again the Elementor team!

**Comment by sellmark** [comment]
It does work perfectly on my iPad + Chrome, but cant perform touch drag actions on the Android device.

Anyone tried on iPhone?

**Comment by 5Diraptor** [comment]
Same, can't edit on Android tablet.  :focus pseudo-selector is recognised but I'm guessing that's because it's the same pseudo selector for touch and non touch.

**Comment by henryperkins** [comment]
I really would love to see this functionality on a mobile / tablet / touch screen device for productivity! Any news on this?

**Comment by markusd1984** [comment]
Lol after so many years still no update, when this could be so easily solved by providing another mechanism to select elements and paste them, i.e. enable a paste option for the selected element in the long press menu (right click on mouse).

**Comment by vogelmy** [comment]
Please add support for tablets!
My client really needs it.

**Comment by OsasDr** [comment]
This is really frustrating. It would have been better this seamless functionality is activated by Elementor. The puffin browser is an encouragement that something good will come

**Comment by ghost** [comment]
Please can you enable drag and drop functionality for elementor for Android devices? Beaver builder does this, which is why I keep running back them and I'm still skeptical about purchasing elementor pro

**Comment by indalodesigns** [comment]
After reading through all of this, and having the same issues. I decided to try puffin browser, changed to desktop view (use hamburger at top right), and selected the mouse option (hamburger again), worked immediately.
It's worth installing, even if you only use it for 'on the fly' changes.

**Comment by geotistics** [comment]
Elementor is the most expensive software I use for creating wordpress websites and it's incredibly disappointing that it doesn't have this basic functionality. I've never needed it, never thought about it, but now I am traveling for the holiday and absolutely need it. I have both my iPad Pro and Surface Pro. Packed light for travel so no mouse or keyboard and I can't make one simple change because of this issue. $4000 worth of mobile tech rendered useless. It's 2021, almost 2022. MOBILE DEVICES ARE EXTREMELY COMMON. Boycotting elementor isn't an option for me, but I won't recommend it to anyone else until this problem is resolved.

**Comment by atar-axis** [comment]
To all commenters: please leave a 👍 on the first post too!

**Comment by ConsultMePL** [comment]
It is just incredible that this request is over 5 years old... There is another long-term request - it's not possible to change the size of background images for sections/columns. It load full size image, which sometimes makes the file size 5-7x as large as needed... Obviously no replies in both threads.

They should take an example from ClickUp, how to deal with customers requests!

**Comment by KensonPlayz** [comment]
This NEEDS to be added. Come on... Tablets, touch screen laptops/monitors, will not be going away at this point (tablets/laptops at least)

**Comment by rushter777** [comment]
Have all of the responders here actually tried? I agreed with this post
when I saw it a few years ago as I had the same frustrations. I now have an
iPad pro 2020 with an apple pencil and I've had to make changes to
Elementor with it. I can see the curve in getting used to it, but saying it
doesn't work at all is incorrect.
First, you should be creating your clients' sites on a desktop or laptop.
Second, set up your clients where the only changes they can do are on the
front end.

On Tue, Jan 4, 2022 at 11:28 KensonPlayz ***@***.***> wrote:

> This NEEDS to be added. Come on... Tablets, touch screen laptops/monitors,
> will not be going away at this point (tablets/laptops at least)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elementor/elementor/issues/482#issuecomment-1004483572>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEFFGO64DEXDSMCQMLYX3VTUUJLNNANCNFSM4COH2ONA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Thanks, Scott Kauer

**Comment by KensonPlayz** [comment]
> Have all of the responders here actually tried? I agreed with this post when I saw it a few years ago as I had the same frustrations. I now have an iPad pro 2020 with an apple pencil and I've had to make changes to Elementor with it. I can see the curve in getting used to it, but saying it doesn't work at all is incorrect. First, you should be creating your clients' sites on a desktop or laptop. Second, set up your clients where the only changes they can do are on the front end.
> On Tue, Jan 4, 2022 at 11:28 KensonPlayz ***@***.***> wrote: This NEEDS to be added. Come on... Tablets, touch screen laptops/monitors, will not be going away at this point (tablets/laptops at least) — Reply to this email directly, view it on GitHub <[#482 (comment)](https://github.com/elementor/elementor/issues/482#issuecomment-1004483572)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEFFGO64DEXDSMCQMLYX3VTUUJLNNANCNFSM4COH2ONA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. You are receiving this because you commented.Message ID: ***@***.***>
> -- Thanks, Scott Kauer

Doesn't work on my Galaxy Tab S7 with or without stylus. Nor my S20+. I've tried Vivaldi, Chrome, Edge, and Puffin browser. None work.

**Comment by RafaelDeJongh** [comment]
I myself am in this state where many of my clients switches to a mobile device like a tablet or phone and I get so many calls/emails on why it is so difficult to even not possible to modify their website when using elementor. At one point I even considered rolling back to an older page builder and Heck maybe even Gutenberg!

I was actually looking if anyone already made an addon to fix this, but I barely found anything other than this problem explenation. 

Yet none could provide me a solution, and for that I do want to for myself and the Elementor community with a third party option that I would create to make these little bugs & snafoos like drag and drop, not being able to scroll to the bottom of the main settings to even press the publish/update page, and in general optimise the editor for both tablets and android phones by utilising different view methods to showcase the panels and what not.

Though this won't be something that is done fast and also requires quite a lot of work to keep it maintained after each elementor update to check if everything still works.

So if a large enough number of members really want this and perhaps want to help in whatever way (your own time to design, code, or pool money together for a release) then I do think this should be possible.

People have been asking for so long now without any imput from Elementor other than: mobile devices are not supported being their standard sentence when you even have editor and mobile device in the same sentence tbh...

So for that reason I'm willing to take this task upon me, and once it is finished I will present this to the elementor devs so they can make it their own.
Therefor if you believe in this idea and want to help me, then do let me know something via here or directly to my company: [Info@Yama.Digital](mailto:Info@Yama.Digital)

If there are enough people that are interested then I'll make a new thread as well where this could be discussed.

**Comment by SwastikBelieves** [comment]
**Yes. We can use elementor with our Android phone.**

Want to know how??

Simply the answer is **Puffin Browser**

I used Puffin Browser and made my whole site with mobile phone. You can visit my site and see : https://www.eruditiontech.in

**Follow these simple steps to edit elementor in mobile** : 

- Download Puffin Browser. It's available in Play Store. I recommend to use it's pro version to use it properly. Simply search "_Puffin Browser pro download_" on Google.
- Go to **settings** page of the app
- Enable Mouce and Keyboard option there
- Login to your WordPress dashboard
- Click on _edit with elementor_ option of your desired page
- You have to move the cursor of your required widgets and **hold for few seconds** to drag it 
- Drop into your page

If you face any problem to customise your page in Puffin Browser, just simply build your site's structure and click on "_Update_". Then come to chorme and customise your pages. 

**ALERT** :  You have to **HOLD YOUR CURSOR ON WIDGETS** for few seconds to drag it

**Comment by brianf012** [comment]
Puffin does not work on the s7+ tablet. In addition, it looks like the pro version is going away and they will start charging monthly. Anyone else find a way to get Elementor widget dragging to work on Android? It doesnt even work with a mouse attached to the tablet. Maybe user agent changing to trick Elementor?

**Comment by imsahiraf** [comment]
Why should we use another browser why Elementor can't add any options like this do to so? Even Pagelayer can be usable in Mobile Phones Elementor should also work on the same.

**Comment by jackkyjohn** [comment]
Six year counting, Elementor is yet to integrate their site to support drag and drop on mobile device

**Comment by androidacy-user** [comment]
The answer none of you want is they're not going to do anything.

They're making their money just fine without adding mobile usability and they don't care about that aspect.

**Comment by imsahiraf** [comment]
> Six year counting, Elementor is yet to integrate their site to support drag and drop on mobile device

I think some other plugin will take its place one day

**Comment by DonvdH** [comment]
Unfortunately the latest version of Elementor doesn't seem to work anymore in combination with Puffin:
https://babiato.co/threads/help-regarding-elementor-on-puffin-browser.76930/

I'm personally also getting the "Preview could not be loaded" error.
Is there anyone that has the latest version of Elementor working with Puffin?

**Comment by joefclarin** [comment]
Created an empty WordPress-Elementor website and just tried Puffin browser on my android phone and the Elementor editor is not loading.

**Comment by toximaberry** [comment]
Lol seems their developer can't achieve that and they are not willing to pay someone that can do that for them 😂, it's either you use a theme that uses Pagelayer or Wpbackery.... Very soon elemetor will be left behind if they refuse to work on their mobile editor because almost every desktop app now has its own mobile version...these days you can literally do everything on your phone .... I've designed more than 20 website and I do everything on my phone using Wpbackery, I just hope Elementor wakes up from their slumber

**Comment by toximaberry** [comment]
I think the only solution to this issue for now is to go to (Elementor > Settings > Features> Top bar). Then activate it, you will be able to choose Element but you won't be able to drg and drop, it also solve save button not displaying on mobile device with smaller screen ... You can checkout my blog for more tech tips toximaspace.com.ng/blog

**Comment by nicholaszein** [comment]
Hi there,

### 📅  This request is scheduled to be Migrated to the new [Discussions](https://github.com/elementor/elementor/discussions/categories/feature-request) section.

Please do not duplicate this request.

Once we migrate it, we encourage you to cast your **⬆️ Upvote** on the first comment if you feel it should be considered for our roadmap.

**Comment by nicholaszein** [comment]
Hi there,

### ✈️ This Feature Request has been migrated from issue #482 

if you still think this request should be considered for our roadmap, please cast your **⬆️ Upvote** on the first comment.

Happy voting! 🙂

**Reply by androidacy-user** [reply]
And you'll actually consider those votes I suppose? Since this issue has been open for over 3 years and it's 2023 and your builder isn't properly working on mobile devices....

**Reply by markusd1984** [reply]
some reassuring would be great, it's a bit sad it took so long just to get some acknowledgement (but better later then never, even 7yrs after), a lot of users switching to Showit!

**Reply by nicholaszein** [reply]
Hi, 

This feature has always been on our radar, and the former comments were all migrated to this discussion. Everything is being taken into consideration.

There are some limitations of mobile browsers (especially Safari) that prevented Elementor from working correctly on some cases, which is the main reason (but not the only one) why Elementor isn't yet compatible with touch devices, although users have been able to edit Elementor pages using a mobile device, keyboard and mouse for quite some time, in most situations. 

One of the conditions for releasing this feature is waiting for mobile technology to improve (and it has gotten better) in order to implement this request. This will mean increasing the browser requirements to even newer versions. 

We will continue considering this request for a future release.

We can't give you an ETA for when this will happen. For the time being or focus is on fixing bugs and improving the overall quality and stability of the editor and all the new features introduced this year's so far.

Please follow our roadmap. That is the best place to know if and when a new feature is being worked on.

https://elementor.com/roadmap

**Reply by androidacy-user** [reply]
So tldr it's not happening but we'll make vague promises of it happening on some ephemeral date in the future and give excuses as to why we couldn't do what almost everyone else has. Gotcha.

**Reply by henryperkins** [reply]
Unlike my desktop browser environment, my mobile browsing experience is not married to Safari (have always used iPhone). In fact, from a developer's perspective, I only use Safari to see if I can interact with the finished product like other mobile browsers. I have no problem switching to Chrome, Edge, or Firefox on mobile to edit. This "especially on Safari" comment is kinda boring.

**Reply by PivitParkour94** [reply]
@nicholaszein I just wanted to say, thank you for the transparency. I believe a lot of my frustration is with No-Code/Low-Code editors generally, as they're making my job less and less relevant (for a certain audience). I'm glad to hear that this is something on your radar and we may see it at some point. Cheers
